File: easygallery-core/src/main/java/za/co/bwmuller/easygallerycore/data/custom/MediaLoader.java
package za.co.bwmuller.easygallerycore.data.custom;

import android.support.annotation.NonNull;

import java.util.ArrayList;

import za.co.bwmuller.easygallerycore.model.Album;
import za.co.bwmuller.easygallerycore.model.Media;

/**
 * Created by Bernhard Müller on 8/24/2017.
 */

public interface MediaLoader {
    @NonNull ArrayList<Media> allMedia();

    @NonNull ArrayList<Media> customAlbumMedia(Album album);
}

File: easygallery-core/src/main/java/za/co/bwmuller/easygallerycore/data/custom/MediaLoader.kt
package za.co.bwmuller.easygallerycore.data.custom

import za.co.bwmuller.easygallerycore.model.Album
import za.co.bwmuller.easygallerycore.model.Media
import java.util.ArrayList

/**
 * Created by Bernhard Müller on 8/24/2017.
 */
interface MediaLoader {

    fun allMedia(): ArrayList<Media>
    fun customAlbumMedia(album: Album): ArrayList<Media>
}